rubicon/objc/types.py

```diff
@@ -90,16 +90,15 @@
 
 __LP64__ = 8 * struct.calcsize("P") == 64
 
-# platform.machine() indicates the machine's physical architecture,
-# which means that on a 64-bit Intel machine it is always "x86_64",
-# even if Python is built as 32-bit.
-_any_x86 = platform.machine() in ("i386", "x86_64")
+# os.uname() provides system identification details; the `machine` attribute
+# indicates the machine's physical architecture. On a 64-bit Intel machine it is
+# always "x86_64", even if Python is built as 32-bit.
+_machine = os.uname().machine
+_any_x86 = _machine in ("i386", "x86_64")
 __i386__ = _any_x86 and not __LP64__
 __x86_64__ = _any_x86 and __LP64__
 
-# On iOS, platform.machine() is a device identifier like "iPhone9,4",
-# but the platform.version() string contains the architecture.
-_any_arm = "ARM" in platform.version()
+_any_arm = _machine.startswith("arm")
```

Comment:
Will this work on actual iOS/iPadOS devices?

From what I can tell, the deleted comment is still accurate as of iOS 15 - the `machine` string returned by the `uname` system call is still a device identifier and not the CPU architecture. The comment you added in beeware/Python-Apple-support@2f42105838ab8f6f7e703ddb929d97758a36145e seems to confirm this.

This means the check will incorrectly fail on real hardware, so the code won't detect *any* architecture. When that happens, it seems that Rubicon by chance does all the right things for ARM64, but we obviously shouldn't rely on that.

Reply:
Dang... you're right. I took the "it works in testing" to be an indication that `os.uname.machine` was doing the right thing. Updated to use `platform.processor()`.